cryptonight/cuda_cryptonight_extra.cu

```diff
@@ -99,6 +100,15 @@ __global__ void cryptonight_extra_gpu_prepare(int threads, const uint32_t * __re
 		XOR_BLOCKS_DST(ctx_state, ctx_state + 4, ctx_a);
 		XOR_BLOCKS_DST(ctx_state + 2, ctx_state + 6, ctx_b);
 
+                if (variant > 0)
+		{
+			tweak1_2[0] = (input[8] >> 24) | (input[9] << 8);
+			tweak1_2[0] ^= (ctx_state[24] & 0xffffffff);
+			tweak1_2[1] = (input[9] >> 24) | (input[10] << 8);
+			tweak1_2[1] ^= (ctx_state[24] >> 32);
```

Comment:
if you shift the 32bit value `ctx_state` by 32 to the right you get zero. Does this mean it is equal to `tweak1_2[1] ^= 0` is equivalent to `do nothing`?